Tests/StarknetTests/Providers/ProviderTests.swift

```diff
@@ -396,10 +399,26 @@ final class ProviderTests: XCTestCase {
             classHash: deployAccountTx.classHash
         )
 
-        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate])
+        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate]).send()
 
         XCTAssertEqual(simulations2.count, 2)
         XCTAssertTrue(simulations2[0].transactionTrace is StarknetInvokeTransactionTrace)
         XCTAssertTrue(simulations2[1].transactionTrace is StarknetDeployAccountTransactionTrace)
     }
+
+    func testBatchGetTransactionByHash() async throws {
+        let previousResult = try await provider.getTransactionBy(blockId: .tag(.latest), index: 0).send()
+
+        let transactionsResponse = try await provider.batchRequests(requests:
+            provider.getTransactionBy(hash: previousResult.transaction.hash!),
+            provider.getTransactionBy(hash: "0x123")).send()
+
+        XCTAssertEqual(transactionsResponse.count, 2)
+        XCTAssertEqual(try transactionsResponse[0].get().transaction.hash, previousResult.transaction.hash)
+
+        do {
+            let _ = try transactionsResponse[1].get().transaction.hash
+            XCTFail("Fetching transaction with nonexistent hash should fail")
+        } catch {}
```

Comment:
We can catch an error and verify if it's correct one

Reply:
Sure, I've added check for error message 👍